apps/storybook.namekit.io/stories/Namekit/utils.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  DomainCard,
+  PrimaryRegistrationStatus,
+  Registration,
+  SecondaryRegistrationStatus,
+  addSeconds,
+  buildDuration,
+  buildENSName,
+  now,
+  subtractSeconds,
+} from "@namehash/ens-utils";
+import { Normalization } from "@namehash/nameguard";
+
+export const REGISTRATION_OF_EXPIRING_SOON_DOMAIN = {
+  registrationTimestamp: subtractSeconds(now(), buildDuration(1000n)),
+  expirationTimestamp: addSeconds(now(), buildDuration(4000n)),
+  expiryTimestamp: null,
+  primaryStatus: PrimaryRegistrationStatus.Active,
+  secondaryStatus: SecondaryRegistrationStatus.ExpiringSoon,
+};
+
+export const REGISTRATION_OF_EXPIRED_DOMAIN = {
+  registrationTimestamp: subtractSeconds(now(), buildDuration(1000n)),
+  expirationTimestamp: addSeconds(now(), buildDuration(4000n)),
+  expiryTimestamp: null,
+  primaryStatus: PrimaryRegistrationStatus.Expired,
+  secondaryStatus: SecondaryRegistrationStatus.GracePeriod,
+};
+
+export const REGISTRATION_OF_NORMAL_EXPIRATION_STATUS = {
+  registrationTimestamp: null,
+  expirationTimestamp: null,
+  expiryTimestamp: null,
+  primaryStatus: PrimaryRegistrationStatus.Active,
+  secondaryStatus: null,
+};
+
+export enum DomainStatus {
+  ExpiringSoon,
+  Expired,
+  Normal,
+}
+
+export type DomainCardOwnerProps = {
+  ownerAddress: `0x${string}` | null;
+  managerAddress: `0x${string}` | null;
+  /** Former owner address is only set when the domain is in Grace Period */
+  formerOwnerAddress: `0x${string}` | null;
+  /** Former manager address is only set when the domain is in Grace Period */
+  formerManagerAddress: `0x${string}` | null;
+};
+
+export enum ENSNameVariant {
+  NormalizedWithAvatar = "NormalizedWithAvatar",
+  NormalizedWithoutAvatar = "NormalizedWithoutAvatar",
+  Unnormalized = "Unnormalized",
+  Unknown = "Unknown",
+}
+
+export const getENSnameFor = (variant: ENSNameVariant) => {
+  switch (variant) {
+    case ENSNameVariant.NormalizedWithAvatar:
+      return buildENSName("lightwalker.eth");
+    case ENSNameVariant.NormalizedWithoutAvatar:
+      return buildENSName("aksjdnaaksdjnsakjdnasd.eth");
+    case ENSNameVariant.Unnormalized:
+      return buildENSName("‍420.eth");
+    case ENSNameVariant.Unknown:
+      return buildENSName(
+        "[c8712e1bff32e40db0e8e57e1b0d0d2b9733a30a9b63d3dfb06fa292a8a17348].eth",
+      );
+  }
+};
+
+export const getMockedDomainCard = ({
+  domainStatus,
+  isOwner = true,
+  normalization = Normalization.normalized,
+}: {
+  isOwner?: boolean;
+  domainStatus: DomainStatus;
+  normalization?: Normalization;
+}): DomainCard => {
+  const address = isOwner
```

Comment:
Hmmm, this is a mock only created for showcasing different domain scenarios...

Whoever uses our library and wants to make usage of components or functions that make usage of `DomainCard` properties should need to create `DomainCard` objects from specific domains they want, which generates the need of making usage of our `getDomainCard` API... Are we planning to make this open-source as well?

Reply:
@FrancoAguzzi Yes, we are definitely working towards making the `getDomainCard` API open source. But this needs more work before it is ready.

When I see these hardcoded addresses named "is owner" and "non owner" it's a big red flag for me that we're not doing this properly.

We're building these UI components for anyone to use. Why do we have our logic defined this way? Is this code being written so that the only way I can see the "is owner" use case is if I am signed in with the wallet address defined by `MOCKED_DOMAIN_CARD_IS_OWNER_ADDRESS`? That's all wrong.

Please ensure there's a clean solution for anyone to test both the is owner and non-owner use cases.

The issue is NOT that some address values are hardcoded. That's fine and I don't care about that. The issue is if only 1 address in the world is able to test the is owner use case then there's a big problem.

Reply:
@lightwalker-eth the addresses that display one or another Ui component use case are all parameters of the Ui component itself, so anyone that uses this Ui component can compare any addresses to generate outputs 👍🏼

Reply:
About hardcoding these values, this was done to replicate DomainCard objects based on the different use cases we wanted to display in our Storybook stories, not being, at all, a constraint to the Ui component exported, itself.

This function was removed since `DomainCard` usage was deprecated in these components, today 👍🏼

Reply:
@FrancoAguzzi All the confusion here was coming from confusing names being given to things. In the latest PR review I've suggested improvements.